src/mlpack/tests/augmented_rnns_tasks_test.cpp

```diff
@@ -0,0 +1,281 @@
+    }
+    size_t tot_len = binary_seq.size();
+    labels = arma::zeros(3, tot_len);
+    for (size_t j = 0; j < tot_len; ++j) {
+      labels.at(binary_seq[tot_len-j-1], j) = 1;
+    }
+    labels.reshape(predictors.n_elem, 1);
+  }
+  void Predict(
+      arma::field<arma::mat>& predictors,
+      arma::field<arma::mat>& labels) {
+    size_t sz = predictors.n_elem;
+    labels = arma::field<arma::mat>(sz);
+    for (size_t i = 0; i < sz; ++i) {
+      Predict(predictors.at(i), labels.at(i));
+    }
+  }
+
+ private:
+  size_t bitLen;
+};
+
+BOOST_AUTO_TEST_SUITE(AugmentedRNNsTasks);
+
+BOOST_AUTO_TEST_CASE(CopyTaskTest)
+{
+  bool ok = true;
+  // Check the setup on vrious lengths...
+  for (size_t maxLen = 2; maxLen <= 16; ++maxLen) {
+    // .. and various numbers of repetitions.
+    for (size_t nRepeats = 1; nRepeats <= 10; ++nRepeats) {
+      CopyTask task(maxLen, nRepeats);
+      arma::field<arma::mat> trainPredictor, trainResponse;
+      task.Generate(trainPredictor, trainResponse, 8);
+      arma::field<arma::mat> testPredictor, testResponse;
+      task.Generate(testPredictor, testResponse, 8);
+      HardCodedCopyModel model;
+      model.Train(trainPredictor, trainResponse);
+      arma::field<arma::mat> predResponse;
+      model.Predict(testPredictor, predResponse);
+      // A single failure is a failure.
+      if (SequencePrecision<arma::mat>(
+            testResponse, predResponse) < 0.99) {
+        ok = false;
+        break;
+      }
+    }
+    if (!ok) break;
+  }
+  BOOST_REQUIRE(ok);
+}
+
+BOOST_AUTO_TEST_CASE(SortTaskTest) {
+  bool ok = true;
+  size_t bitLen = 5;
+  for (size_t maxLen = 2; maxLen <= 16; ++maxLen) {
+    SortTask task(maxLen, bitLen);
+    arma::field<arma::mat> trainPredictor, trainResponse;
+    task.Generate(trainPredictor, trainResponse, 8);
+    arma::field<arma::mat> testPredictor, testResponse;
+    task.Generate(testPredictor, testResponse, 8);
+    HardCodedSortModel model(bitLen);
+    model.Train(trainPredictor, trainResponse);
+    arma::field<arma::mat> predResponse;
+    model.Predict(testPredictor, predResponse);
+    // A single failure is a failure.
+    if (SequencePrecision<arma::mat>(testResponse, predResponse) < 0.99) {
+      ok = false;
+      break;
+    }
+  }
+
+  BOOST_REQUIRE(ok);
+}
+
+BOOST_AUTO_TEST_CASE(AddTaskTest) {
+  bool ok = true;
+  for (size_t bitLen = 2; bitLen <= 16; ++bitLen) {
+    AddTask task(bitLen);
+    arma::field<arma::mat> trainPredictor, trainResponse;
+    task.Generate(trainPredictor, trainResponse, 8);
+    arma::field<arma::mat> testPredictor, testResponse;
+    task.Generate(testPredictor, testResponse, 8);
+    HardCodedAddModel model;
+    model.Train(trainPredictor, trainResponse);
+    arma::field<arma::mat> predResponse;
+    model.Predict(testPredictor, predResponse);
+    // A single failure is a failure.
+    if (SequencePrecision<arma::mat>(testResponse, predResponse) < 0.99) {
+      ok = false;
+      break;
+    }
```

Comment:
No need to break, you can just check like this:

```
BOOST_REQUIRE_GE(SequencePrecision<arma::mat>(testResponse, predResponse), 0.99);
```

and it will terminate the test if there are any failures at all.  Another nice thing about this, is that it will show what the precision actually was instead of just `true` or `false` like the `BOOST_REQUIRE()` test.